src/scratchtocatrobat/scratch/scratch.py

```diff
@@ -521,36 +543,35 @@ def _add_sensor_variables_and_update_script_to_object(self, sprite_object, senso
         from scratchtocatrobat.converter import converter
         forever_loop_body_blocks = []
         sprite_name = sprite_object.get_objName()
-        for command, param in sensors_info:
+        for info in sensors_info:
+            command = info["cmd"]
+            param = info["param"]
             if not converter.is_supported_block(command) and command != "timer":
                 continue
 
-            if sprite_name not in self.sprite_variables_map:
-                self.sprite_variables_map[sprite_name] = []
-
             stage_object = self.objects[0]
             if command == "timer":
                 variable_name = S2CC_TIMER_VARIABLE_NAME
-                self.sprite_variables_map[sprite_name] += [variable_name]
                 if not is_add_timer_script:
                     self._add_timer_script_to_stage_object()
-                continue
             elif command == "answer":
                 variable_name = converter._SHARED_GLOBAL_ANSWER_VARIABLE_NAME
-                self.sprite_variables_map[sprite_name] += [variable_name]
                 stage_object._dict_object["variables"].append({ "name": variable_name, "value": "", "isPersistent": False })
-                continue
-
-            variable_name = S2CC_SENSOR_PREFIX + "{}_{}{}".format(sprite_name, command, "_" + param if param else "")
-            self.sprite_variables_map[sprite_name] += [variable_name]
-            sprite_object._dict_object["variables"].append({ "name": variable_name, "value": 0, "isPersistent": False })
-            reporter_block = [command] if param is None else [command, param]
-            forever_loop_body_blocks += [["setVar:to:", variable_name, reporter_block]]
-
-        if len(forever_loop_body_blocks) == 0: return
-        forever_loop_body_blocks += [["wait:elapsed:from:", UPDATE_HELPER_VARIABLE_TIMEOUT]]
-        script_blocks = [["doForever", forever_loop_body_blocks]]
-        sprite_object.scripts += [Script([0, 0, [[SCRIPT_GREEN_FLAG]] + script_blocks])]
+            else:
+                variable_name = S2CC_SENSOR_PREFIX + "{}_{}{}".format(sprite_name, command, "_" + param if param else "")
```

Comment:
In a test with the backdrop number being monitored in the scratch project, the conversion crashed when creating the variable name because param actually is a dictionary in this case. 
The background number seems not to be converted correctly at the moment so preventing the crash should be enough here.
![Bildschirmfoto 2020-05-19 um 14 09 46](https://user-images.githubusercontent.com/39565761/82324860-9e8a3780-99da-11ea-9d5a-5bfb4214facf.png)

Reply:
The problem was in the Scratch3 -> Scratch2 conversion, because they merged the block for the background number and name and hat number or name in the params field.
I made a special case for that block and now both blocks work as expected.